apps/s_server.c

```diff
@@ -3340,7 +3340,7 @@ static int www_body(int s, int stype, int prot, unsigned char *context)
             }
             /* send the file */
 #ifndef OPENSSL_NO_KTLS
-            if (use_sendfile) {
+            if (use_sendfile && BIO_get_ktls_send(SSL_get_wbio(con))) {
```

Comment:
I was about to implement this, but here's a thought:

Since whether KTLS can be used depends on e.g. TLS extensions negotiated, I would imagine this varies between individual SSL connections. It would be easy to do a warning which is printed one time only the first time a non-sendfile request is made, but since this can presumably be controlled by the remote party potentially negotiating extensions which will prevent KTLS use, it's not necessarily very useful to the person running `s_server`, and may happen an arbitrarily long time after `s_server` is started.

So there are two options here:

  - Warn on first non-sendfile connection, despite the caveats above;
  - Try and predict whether KTLS will be unavailable before accepting the first connection.

Reply:
Can't we print a warning on every connection? `s_server` is a test tool and not really intended for very long running scenarios - so I'm not sure I agree that its not useful for the person running `s_server`.  Quite the opposite.